### PR TITLE
Implementation of Peng-Robinson EoS (Revised)

### PR DESCRIPTION
**Pull request by gkogekar**
This is a more cleaned up version of the Peng-Robinson EoS implementation. 
(Original PR: https://github.com/Cantera/cantera/pull/641).

### CODE REVIEW

File: src/thermo/PengRobinson.cpp [hunk 1]
```diff
@@ -0,0 +1,817 @@
...
+        sqtTr = sqrt(temperature() / Tc); //we need species critical temperature
+        coeff1 = 1 / (Tc*sqtTr);
+        coeff2 = sqtTr - 1;
+        k = m_kappa[i];
+        m_dalphadT[i] = coeff1 * (k*k*coeff2 - k);
+    }
+    //Calculate mixture derivative
+    for (size_t i = 0; i < m_kk; i++) {
+        for (size_t j = 0; j < m_kk; j++) {
+            temp = 0.5 * sqrt((m_a_coeffs(i, i) * m_a_coeffs(j, j)) / (m_alpha[i] * m_alpha[j]));
```
Comment (decaluwe) [inline src/thermo/PengRobinson.cpp]: Not sure this will impact the failing tests, but you are assuming the user has not provided an `a_ij`, and automatically reverting to the default mixing rules.  This should be:

```suggestion
            temp = 0.5 * m_a_coeffs(i,j) * sqrt(1 / (m_alpha[i] * m_alpha[j]));
```

File: src/thermo/PengRobinson.cpp [hunk 2]
```diff
@@ -0,0 +1,817 @@
...
+            daAlphadT += moleFractions_[i] * moleFractions_[j] * temp
+                        * (m_dalphadT[j] * m_alpha[i] + m_dalphadT[i] * m_alpha[j]);
```
Comment (decaluwe) [inline src/thermo/PengRobinson.cpp]: In general, your derivation document points to a slightly simpler implementation (but it is functionally equivalent):

```suggestion
                       daAlphadT += moleFractions_[i] * moleFractions_[j] * 0.5 * m_aAlpha_binary(i, j) 
                                             * (m_dalphadT[i] / m_alpha[i] + m_dalphadT[j] / m_alapha[j]
```

File: src/thermo/PengRobinson.cpp [hunk 3]
```diff
@@ -0,0 +1,817 @@
...
+    }
+
+    //Calculate mixture derivative
+    double d2aAlphadT2 = 0.0;
+    for (size_t i = 0; i < m_kk; i++) {
+        double alphai = m_alpha[i];
+        for (size_t j = 0; j < m_kk; j++) {
+            double alphaj = m_alpha[j];
+            double alphaij = alphai * alphaj;
+            double temp = 0.5 * sqrt(m_a_coeffs(i, i) * m_a_coeffs(j, j) / (alphaij));
```
Comment (decaluwe) [inline src/thermo/PengRobinson.cpp]: Here also, this ought to at least be:

```suggestion
            double temp = 0.5 * m_a_coeffs(i, j) / sqrt(alphaij));
```
But I also think there is a slightly simpler implementation, which I'll suggest further below.

File: src/thermo/PengRobinson.cpp [hunk 4]
```diff
@@ -0,0 +1,817 @@
...
+        //  Calculate first and second derivatives of alpha for individual species
+        double k = m_kappa[i];
+        m_dalphadT[i] = coeff1 * (k*k*coeff2 - k);
+        m_d2alphadT2[i] = (k*k + k) * coeff1 / (2*sqt_Tr*sqt_Tr*Tcrit_i);
+    }
+
+    //Calculate mixture derivative
+    double d2aAlphadT2 = 0.0;
+    for (size_t i = 0; i < m_kk; i++) {
+        double alphai = m_alpha[i];
```
Comment (decaluwe) [inline src/thermo/PengRobinson.cpp:764]: Is there a reason to store `alphai` and `alphaj` locally, like this, instead of ust calling `m_alpha[i]` and `m_alpha[j]`, respectively, when they are needed?

File: src/thermo/PengRobinson.cpp [hunk 5]
```diff
@@ -0,0 +1,817 @@
...
+        for (size_t j = 0; j < m_kk; j++) {
+            double alphaj = m_alpha[j];
+            double alphaij = alphai * alphaj;
+            double temp = 0.5 * sqrt(m_a_coeffs(i, i) * m_a_coeffs(j, j) / (alphaij));
+            double num = (m_dalphadT[j] * alphai + m_dalphadT[i] * alphaj);
+            double fac1 = -(0.5 / alphaij) * num * num;
+            double fac2 = alphaj * m_d2alphadT2[i] + alphai * m_d2alphadT2[j] + 2. * m_dalphadT[i] * m_dalphadT[j];
+            d2aAlphadT2 += moleFractions_[i] * moleFractions_[j] * temp * (fac1 + fac2);
```
Comment (decaluwe) [inline src/thermo/PengRobinson.cpp]:
```suggestion
            double term1 = m_d2alphadT2[i] / alphai + m_d2alphadT2[j] / alphaj 
            double term2 = 2 * m_dalphadT[i] * m_dalphadT[j] / alphaij 
            double term3 = m_dalphadT[i] / alphai + m_dalphadT[j] / alphaj
            d2aAlphadT2 += 0.5 * moleFractions_[i] * moleFractions_[j] * m_aAlpha_binary(i, j)
                                       * (term1 + term2 + 0.5 * term3 * term3)
```
This is a few lines shorter, but not _that_ much simpler.  But, given the failing tests, it might be worth trying it out.

File: src/thermo/PengRobinson.cpp [hunk 6]
```diff
@@ -0,0 +1,817 @@
...
+
+void PengRobinson::calculateAB(double& aCalc, double& bCalc, double& aAlphaCalc) const
+{
+    bCalc = 0.0;
+    aCalc = 0.0;
+    aAlphaCalc = 0.0;
+    for (size_t i = 0; i < m_kk; i++) {
+        bCalc += moleFractions_[i] * m_b_coeffs[i];
+        for (size_t j = 0; j < m_kk; j++) {
+            double a_vec_Curr = m_a_coeffs(i, j);
```
Comment (decaluwe) [inline src/thermo/PengRobinson.cpp]: Is this variable necessary?  Is there any advantage, over just using `m_a_coeffs(i, j)` directly in the next line?

File: src/thermo/PengRobinson.cpp [hunk 7]
```diff
@@ -0,0 +1,817 @@
...
+
+void PengRobinson::calculatePressureDerivatives() const
+{
+    double T = temperature();
+    double mv = molarVolume();
+    double pres;
+
+    m_dpdV = dpdVCalc(T, mv, pres);
+    double vmb = mv - m_b;
+    double den = mv * mv + 2 * mv * m_b - m_b * m_b;
```
Comment (decaluwe) [inline src/thermo/PengRobinson.cpp]: Might be clearer to call this `denom` for "denominator"... `den` is easy to confuse with "density."

File: src/thermo/PengRobinson.cpp [hunk 8]
```diff
@@ -0,0 +1,817 @@
...
+    std::pair<double, double> vv = bmt::toms748_solve(
+        resid, Vroot[1], Vroot[2], bmt::eps_tolerance<double>(48), maxiter);
+
+    double mmw = meanMolecularWeight();
+    return mmw / (0.5 * (vv.first + vv.second));
+}
+
+double PengRobinson::dpdVCalc(double T, double molarVol, double& presCalc) const
+{
+    double den = molarVol * molarVol + 2 * molarVol * m_b - m_b * m_b;
```
Comment (decaluwe) [inline src/thermo/PengRobinson.cpp]: Here also, `denom` might be less easily confused with "density".

File: src/thermo/PengRobinson.cpp [hunk 9]
```diff
@@ -0,0 +1,817 @@
...
+    presCalc = GasConstant * T / (molarVol - m_b) - m_aAlpha_mix / den;
```
Comment (decaluwe) [inline src/thermo/PengRobinson.cpp]: Is this line used?

**Comment by speth** [comment]
I'd like to suggest _not_ using the button to accept and commit suggested changes - it really generates a cluttered commit history.

File: src/thermo/PengRobinson.cpp (hunk 6 above)
Comment (gkogekar) [reply src/thermo/PengRobinson.cpp]: Corrected.

File: src/thermo/PengRobinson.cpp (hunk 7 above)
Comment (gkogekar) [reply src/thermo/PengRobinson.cpp]: Corrected.

File: src/thermo/PengRobinson.cpp (hunk 8 above)
Comment (gkogekar) [reply src/thermo/PengRobinson.cpp]: Corrected.

File: src/thermo/PengRobinson.cpp (hunk 9 above)
Comment (gkogekar) [reply src/thermo/PengRobinson.cpp]: I have removed this line. It wasn't used anywhere.

File: data/inputs/critProperties.xml [hunk 10]
```diff
@@ -76,7 +76,7 @@
     <Pc value="7.39e+06"/>
     <Vc value="0.0948"/>
     <Zc value="0.275"/>
-    <omega value="0.239"/>
+    <omega value="0.228"/>
```
Comment (bryanwweber) [inline data/inputs/critProperties.xml:79]: Please add a reference to a source here.

File: include/cantera/thermo/PengRobinson.h [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+
+    //! Vector of derivatives of pressure with respect to mole number
+    /*!
+     *  Calculated at the current conditions. Total volume, temperature and
+     *  other mole number kept constant
+     */
+    mutable vector_fp m_dpdni;
+
+private:
+    //! Omega constants: a0 (= omega_a) and b0 (= omega_b) values used in Peng-Robinson equation of state
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]: This comment got associated with the `omega_a` documentation, but it doesn't seem to be related just to that one. Can you rephrase it?

File: include/cantera/thermo/PengRobinson.h [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+     * \f]
+     *
+     *  and
+     *
+     * \f[
+     *    \kappa = \left(0.37464 + 1.54226\omega - 0.26992\omega^2\right), \qquad \qquad \text{For } \omega <= 0.491 \\
+     *    \kappa = \left(0.379642 + 1.487503\omega - 0.164423\omega^2 +  0.016667\omega^3 \right), \qquad \text{For } \omega > 0.491
+     * \f]
+     *
+     * Coefficients \f$ a_mix, b_mix \f$ and \f$(a \alpha)_{mix}\f$ are calculated as
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
     * Coefficients \f$ a_{mix}, b_{mix} \f$ and \f$(a \alpha)_{mix}\f$ are calculated as
```

File: include/cantera/thermo/PengRobinson.h [hunk 13]
```diff
@@ -0,0 +1,313 @@
...
+    virtual void getActivityCoefficients(double* ac) const;
+
+    /// @name  Partial Molar Properties of the Solution
+    //@{
+
+    virtual void getChemPotentials(double* mu) const;
+    virtual void getPartialMolarEnthalpies(double* hbar) const;
+    virtual void getPartialMolarEntropies(double* sbar) const;
+    virtual void getPartialMolarIntEnergies(double* ubar) const;
+    virtual void getPartialMolarCp(double* cpbar) const;
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h:127]: Can you indicate in the documentation that this function is not implemented?

File: include/cantera/thermo/PengRobinson.h [hunk 14]
```diff
@@ -0,0 +1,313 @@
...
+    virtual void getPartialMolarVolumes(double* vbar) const;
+
+    //! Calculate species-specific critical temperature
+    /*!
+     *  The temperature dependent parameter in P-R EoS is calculated as
+     *       \f$ T_{crit} = (0.0778 a)/(0.4572 b R) \f$
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
     *       \f[ T_{crit} = (0.0778 a)/(0.4572 b R) \f]
```

File: include/cantera/thermo/PengRobinson.h [hunk 15]
```diff
@@ -0,0 +1,313 @@
...
+     * the phase and setting its parameters from a specification in an
+     * input file. They are not normally used in application programs.
+     * To see how they are used, see importPhase().
+     */
+    //@{
+
+    virtual bool addSpecies(shared_ptr<Species> spec);
+    virtual void initThermo();
+
+    //! Retrieve a and b coefficients by looking up tabulated critical parameters
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    //! Retrieve \f$a\f$ and \f$b\f$ coefficients by looking up tabulated critical parameters
```

File: include/cantera/thermo/PengRobinson.h [hunk 16]
```diff
@@ -0,0 +1,313 @@
...
+    /*!
+    *  If pureFluidParameters are not provided for any species in the phase,
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    *  If `pureFluidParameters` are not provided for any species in the phase,
```

File: include/cantera/thermo/PengRobinson.h [hunk 17]
```diff
@@ -0,0 +1,313 @@
...
+    *  consult the critical properties tabulated in data/inputs/critProperties.xml.
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    *  consult the critical properties tabulated in `data/inputs/critProperties.xml`.
```

File: include/cantera/thermo/PengRobinson.h [hunk 18]
```diff
@@ -0,0 +1,313 @@
...
+    *  If the species is found there, calculate pure fluid parameters a_k and b_k as:
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    *  If the species is found there, calculate pure fluid parameters \f$a_k\f$ and \f$b_k\f$ as:
```

File: include/cantera/thermo/PengRobinson.h [hunk 19]
```diff
@@ -0,0 +1,313 @@
...
+    *  \f[ b_k = 0.08664 R T_c/ P_c \f]
+    *
+    *  @param iName    Name of the species
+    */
+    virtual vector_fp getCoeff(const std::string& iName);
+
+    //! Set the pure fluid interaction parameters for a species
+    /*!
+     *  @param species   Name of the species
+     *  @param a         "a" parameter in the Peng-Robinson model [Pa-m^6/kmol^2]
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
     *  @param a         \f$a\f$ parameter in the Peng-Robinson model [Pa-m^6/kmol^2]
```

File: include/cantera/thermo/PengRobinson.h [hunk 20]
```diff
@@ -0,0 +1,313 @@
...
+     *  @param b         "b" parameter in the Peng-Robinson model [m^3/kmol]
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
     *  @param b         \f$b\f$ parameter in the Peng-Robinson model [m^3/kmol]
```

File: include/cantera/thermo/PengRobinson.h [hunk 21]
```diff
@@ -0,0 +1,313 @@
...
+     *  @param w         acentric factor
+     */
+    void setSpeciesCoeffs(const std::string& species, double a, double b,
+                          double w);
+
+    //! Set values for the interaction parameter between two species
+    /*!
+     *  @param species_i   Name of one species
+     *  @param species_j   Name of the other species
+     *  @param a           constant term in the "a" expression [Pa-m^6/kmol^2]
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
     *  @param a           constant term in the \f$a\f$ expression [Pa-m^6/kmol^2]
```

File: include/cantera/thermo/PengRobinson.h [hunk 22]
```diff
@@ -0,0 +1,313 @@
...
+     *  @param alpha       dimensionless function of T_r and \omega
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
     *  @param alpha       dimensionless function of \f$T_r\f$ and \f$\omega\f$
```

File: include/cantera/thermo/PengRobinson.h [hunk 23]
```diff
@@ -0,0 +1,313 @@
...
+    virtual double liquidVolEst(double TKelvin, double& pres) const;
+    virtual double densityCalc(double TKelvin, double pressure, int phase, double rhoguess);
+
+    virtual double densSpinodalLiquid() const;
+    virtual double densSpinodalGas() const;
+    virtual double dpdVCalc(double TKelvin, double molarVol, double& presCalc) const;
+
+    // Special functions not inherited from MixtureFugacityTP
+
+    //! Calculate temperature derivative d(a*alpha)/dT
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    //! Calculate temperature derivative \f$d(a \alpha)/dT\f$
```

File: include/cantera/thermo/PengRobinson.h [hunk 24]
```diff
@@ -0,0 +1,313 @@
...
+    /*!
+     *  These are stored internally.
+     */
+    double daAlpha_dT() const;
+
+    //! Calculate second derivative d2(a*alpha)/dT2
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    //! Calculate second derivative \f$d^2(a \alpha)/dT^2\f$
```

File: include/cantera/thermo/PengRobinson.h [hunk 25]
```diff
@@ -0,0 +1,313 @@
...
+    /*!
+     *  These are stored internally.
+     */
+    double d2aAlpha_dT2() const;
+
+public:
+
+    //! Calculate dpdV and dpdT at the current conditions
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    //! Calculate \f$dp/dV\f$ and \f$dp/dT\f$ at the current conditions
```

File: include/cantera/thermo/PengRobinson.h [hunk 26]
```diff
@@ -0,0 +1,313 @@
...
+    /*!
+     *  These are stored internally.
+     */
+    void calculatePressureDerivatives() const;
+
+    //! Update the a, b and alpha parameters
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    //! Update the \f$a\f$, \f$b\f$, and \f$\alpha\f$ parameters
```

File: include/cantera/thermo/PengRobinson.h [hunk 27]
```diff
@@ -0,0 +1,313 @@
...
+    /*!
+     *  The a and the b parameters depend on the mole fraction and the
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
     *  The \f$a\f$ and the \f$b\f$ parameters depend on the mole fraction and the
```

File: include/cantera/thermo/PengRobinson.h [hunk 28]
```diff
@@ -0,0 +1,313 @@
...
+     *  parameter alpha depends on the temperature. This function updates
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
     *  parameter $\f\alpha\f$ depends on the temperature. This function updates
```

File: include/cantera/thermo/PengRobinson.h [hunk 29]
```diff
@@ -0,0 +1,313 @@
...
+     *  the internal numbers based on the state of the object.
+     */
+    virtual void updateMixingExpressions();
+
+    //! Calculate the a, b and the alpha parameters given the temperature
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    //! Calculate the \f$a\f$, \f$b\f$, and \f$\alpha\f$ parameters given the temperature
```

File: include/cantera/thermo/PengRobinson.h [hunk 30]
```diff
@@ -0,0 +1,313 @@
...
+     */
+    void calculateAB(double& aCalc, double& bCalc, double& aAlpha) const;
+
+    void calcCriticalConditions(double& pc, double& tc, double& vc) const;
+
+    //! Prepare variables and call the function to solve the cubic equation of state
+    int solveCubic(double T, double pres, double a, double b, double aAlpha,
+                   double Vroot[3]) const;
+protected:
+    //! Value of b in the equation of state
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    //! Value of \f$b\f$ in the equation of state
```

File: include/cantera/thermo/PengRobinson.h [hunk 31]
```diff
@@ -0,0 +1,313 @@
...
+    /*!
+     *  m_b is a function of the mole fractions and species-specific b values.
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
     *  `m_b` is a function of the mole fractions and species-specific \f$b\f$ values.
```

File: include/cantera/thermo/PengRobinson.h [hunk 32]
```diff
@@ -0,0 +1,313 @@
...
+     */
+    double m_b;
+
+    //! Value of a and alpha in the equation of state
+    /*!
+     *  m_aAlpha_mix is a function of the temperature and the mole fractions. m_a depends only on the mole fractions.
+     */
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    //! Value of \f$a\f$ and \f$\alpha\f$ in the equation of state
    /*!
     *  m_aAlpha_mix is a function of the temperature and the mole fractions. m_a depends only on the mole fractions.
     */
```

Can you please split these docstrings? `m_aAlpha_mix` doesn't show up in the docs right now.

File: include/cantera/thermo/PengRobinson.h [hunk 33]
```diff
@@ -0,0 +1,313 @@
...
+    double m_aAlpha_mix;
+
+    // Vectors required to store a_coeff, b_coeff, alpha, kappa and other values for every species. Length = m_kk
+    vector_fp m_b_coeffs;
+    vector_fp m_kappa;
+    mutable vector_fp m_dalphadT;
+    mutable vector_fp m_d2alphadT2;
+    vector_fp m_alpha;
+
+    //Matrices for Binary coefficients a_{i,j} and {a*alpha}_{i.j} are saved in an Array form. Length =  (m_kk, m_kk)
```
Comment (bryanwweber) [inline include/cantera/thermo/PengRobinson.h]:
```suggestion
    // Matrices for Binary coefficients a_{i,j} and {a*alpha}_{i.j} are saved in an Array form. Length =  (m_kk, m_kk)
```

I think this line is too long as well.

File: src/thermo/PengRobinson.cpp [hunk 34]
```diff
@@ -0,0 +1,773 @@
...
+            "Species '{}' is not present in the database", iName);
+    }
+    return spCoeff;
+}
+
+void PengRobinson::initThermo()
+{
+    for (auto& item : m_species) {
+        // Read a and b coefficients and acentric factor w_ac from species 'input'
+        // information (i.e. as specified in a YAML input file)
```
Comment (bryanwweber) [inline src/thermo/PengRobinson.cpp]: Why is `'input'` in quotes here? We're also trying to avoid the use of Latin abbreviations such as `i.e.` in new code, since they may not be familiar to everyone.

**Review by bryanwweber — CHANGES_REQUESTED** [review]
Just a few formatting comments here, the biggest one is a source for the value of `omega` in the `critProperties.xml` file.

**Comment by bryanwweber** [comment]
Please also add yourself to the `AUTHORS` file! 🎉

File: data/inputs/critProperties.xml (hunk 10 above)
Comment (gkogekar) [reply data/inputs/critProperties.xml:79]: I have added the reference for this value.

File: include/cantera/thermo/PengRobinson.h (hunk 11 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 12 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 14 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 15 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 16 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 17 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 18 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 19 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 20 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 21 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 22 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 23 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 24 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 25 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 26 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 27 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 28 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 29 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 30 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 31 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 32 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 33 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h]: Corrected.

File: src/thermo/PengRobinson.cpp (hunk 34 above)
Comment (gkogekar) [reply src/thermo/PengRobinson.cpp]: Corrected.

File: include/cantera/thermo/PengRobinson.h (hunk 13 above)
Comment (gkogekar) [reply include/cantera/thermo/PengRobinson.h:127]: I have added a comment to indicate that it is not yet implemented for P-R phase.

**Review by bryanwweber — APPROVED** [review]
Great work here @gkogekar! Thanks @speth and @decaluwe for the detailed reviews as well.

**Comment by decaluwe** [comment]
:tada:
